opentracing/process_recorder.go

```diff
@@ -0,0 +1,26 @@
+package opentracing
+
+// A ProcessRecorder handles all of the `RawSpan` data generated via an
+// associated `OpenTracer` (see `NewStandardTracer`) instance. It also names
+// the containing process and provides access to a straightforward tag map.
+type ProcessRecorder interface {
+	// Every process in the larger distributed system must have a name. It is
+	// up to the `ProcessRecorder` implementation to determine how this name is
+	// specified.
+	ProcessName() string
```

Comment:
see comment above about effectively requiring a process name when constructing an implementation for this interface